cdap-common/src/main/java/co/cask/cdap/common/namespace/AbstractNamespaceClient.java

```diff
@@ -135,7 +135,9 @@ public void deleteDatasets(Id.Namespace namespaceId) throws Exception {
     if (response.getResponseCode() == HttpURLConnection.HTTP_NOT_FOUND) {
       throw new NamespaceNotFoundException(namespaceId);
     } else if (HttpURLConnection.HTTP_FORBIDDEN == response.getResponseCode()) {
-      throw new NamespaceCannotBeDeletedException(namespaceId, response.getResponseBodyAsString());
+      String msg = String.format("Datasets in the namespace '%s' cannot be deleted. ", namespaceId)
```

Comment:
Maybe add a ``Reason: `` after ``deleted.``. Also, you can reuse the existing ``String.format`` instead of concatenating the ``response.getResponseBodyAsString()``